### PR TITLE
Added Create Campaign Modal and also fixes the Auth UI Box size

### PR DESCRIPTION
**Pull request by Manan-Sharma-5**
Screenshot 2024-08-07 at 4 29 36 PM
Screenshot 2024-08-07 at 4 29 25 PM
Screenshot 2024-08-07 at 4 29 17 PM

## Summary by CodeRabbit

- **New Features**
	- Enhanced authentication flow in the `AuthLayout` and `CampaignsLayout` components with dynamic redirection based on user session status.
	- Introduced a multi-step modal `CreateCampaign` component for improved campaign creation experience.
	- Added validation schemas for campaign data to ensure data integrity during the creation process.
	- Implemented a user interface for managing and displaying an API key for OpenAI services.
	- Added a new navigation option for "OpenAI" in the sidebar menu.

- **Bug Fixes**
	- Resolved an error in the `forgot-password` submission flow, allowing for smoother user experience.

- **Style**
	- Improved styling for various components, increasing padding and enhancing layout responsiveness across the app.

- **Chores**
	- Updated import statements and overall code consistency for future enhancements.
	- Added a new dependency for image picking functionality in the project.

### CODE REVIEW

**Comment by Manan-Sharma-5** [comment]
[Compaign -> Create Page](https://trello.com/c/t2d0xYyT/6-compaign-create-page)

File: components/modals/CampaignCreate.tsx
Comment (rahul-sinha1908) [inline components/modals/CampaignCreate.tsx:1]: Split the file in three steps.

**Comment by rahul-sinha1908** [comment]
@coderabbitai review

**Review by rahul-sinha1908 — CHANGES_REQUESTED** [review]

**Comment by Manan-Sharma-5** [comment]
[All routing URLs should be small letters](https://trello.com/c/piTzHje9/6-all-routing-urls-should-be-small-letters)

**Comment by Manan-Sharma-5** [comment]
[Navigating from signup to Login is having issues](https://trello.com/c/orSr00xI/5-navigating-from-signup-to-login-is-having-issues)

**Comment by Manan-Sharma-5** [comment]
[Expand the Central part - Login, Register, ForgotPassword](https://trello.com/c/DTFeTQsi/3-expand-the-central-part-login-register-forgotpassword)

**Comment by Manan-Sharma-5** [comment]
[Login/Signup Pages](https://trello.com/c/2qbeo6RF/3-login-signup-pages)

**Comment by Manan-Sharma-5** [comment]
[Open AI - Add key](https://trello.com/c/SGsdi3n4/14-open-ai-add-key)

**Review by rahul-sinha1908 — APPROVED** [review]

**Pull request merged**